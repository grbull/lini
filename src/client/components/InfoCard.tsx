import cn from 'classnames';
import React, { ReactElement, ReactNode } from 'react';

interface Props {
  children: ReactNode;
  className?: string;
}

export function InfoCard({ children, className }: Props): ReactElement {
  return (
    <div
      className={cn(
        'm-2',
        'px-2',
        'py-2.5',
        'bg-gray-900',
        'bg-opacity-80',
        'overflow-hidden',
        className
      )}
    >
      {children}
    </div>
  );
}

InfoCard.defaultProps = { className: undefined };
