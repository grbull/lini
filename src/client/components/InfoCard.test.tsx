/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { InfoCard } from './InfoCard';

describe('InfoCard Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<InfoCard>Generic children</InfoCard>);
    expect(asFragment()).toMatchSnapshot();
  });
});
