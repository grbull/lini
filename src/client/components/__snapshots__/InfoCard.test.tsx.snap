// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-2 px-2 py-3 bg-gray-200 dark:bg-gray-900 bg-opacity-90 dark:bg-opacity-80 overflow-hidden"
  >
    Generic children
  </div>
</DocumentFragment>
`;
